src/test/scala/chiselTests/QueueFlushSpec.scala

```diff
@@ -0,0 +1,91 @@
+package chiselTests
+
+import org.scalacheck._
+
+import chisel3._
+import chisel3.testers.BasicTester
+import chisel3.util._
+import chisel3.util.random.LFSR
+
+class ThingsPassThroughFlushQueueTester(elements: Seq[Int], queueDepth: Int, bitWidth: Int, tap: Int, useSyncReadMem: Boolean) extends BasicTester {
```

Comment:
please add a test of what happens if you try to write same cycle as trying to flush (should not work) and if you try to read same cycle as flush (should be OK, i think?)